src/V3LinkParse.cpp

```diff
@@ -133,10 +133,24 @@ class LinkParseVisitor final : public VNVisitor {
             VL_RESTORER(m_lifetime);
             {
                 m_ftaskp = nodep;
-                m_lifetime = nodep->lifetime();
-                if (m_lifetime.isNone()) {
-                    // Automatic always until we support static
-                    m_lifetime = VLifetime::AUTOMATIC;
+                if (nodep->lifetime().isNone()) {
+                    const AstClassOrPackageRef* const classPkgRefp
+                        = VN_AS(nodep->classOrPackagep(), ClassOrPackageRef);
+                    if (classPkgRefp && VN_IS(classPkgRefp->classOrPackageNodep(), Class)) {
+                        // Class methods are automatic by default
+                        m_lifetime = VLifetime::AUTOMATIC;
+                    } else if (nodep->dpiImport()) {
+                        // DPI-imported function don't have lifetime specifiers
+                        m_lifetime = VLifetime::NONE;
+                    }
+                    if (m_lifetime.isStatic()
+                        && !nodep->fileline()->warnIsOff(V3ErrorCode::IMPLSTATICFUNC)) {
```

Comment:
```suggestion
                    if (m_lifetime.isStatic()) {
```

v3warn checks for warnIsOff itself.